docs/features/preemption.md

```diff
@@ -0,0 +1,435 @@
+# Quickstart - Preemption
+
+Within the Gödel Scheduling System, Pod preemption emerges as a key feature designed to uphold optimal resource utilization.
+If a Pod cannot be scheduled, the Gödel scheduler tries to preempt (evict) lower priority Pods to make scheduling of the pending Pod possible.
+
+This is a Quickstart guide that will walk you through how preemption works in the Gödel Scheduling System.
+
+## Local Cluster Bootstrap & Installation
+
+Please refer to [Cluster Setup Guide](kind-cluster-setup.md) for creating a local KIND cluster installed with Gödel.
+
+## How Preemption Works
+
+When there is no more resource for scheduling a pending Pod, preemption comes into picture.
+Gödel scheduler tries to preempt (evict) lower priority Pods to make scheduling of the pending Pod possible, with a few of protection strategies being respected.
+
+### Quickstart Scenario
+
+To better illustrate the preemption features, let's assume there is one node with less than 4 CPU cores (actually 3.9 cores) available for scheduling.  
+
+```bash
+$ kubectl describe node kind-worker   
+  Name:               kind-worker
+  ...
+  Capacity:
+    cpu:                4
+    ...
+  Allocatable:
+    cpu:                4
+    ...
+  ...
+  Non-terminated Pods:          (2 in total)
+    Namespace                   Name                CPU Requests  CPU Limits  Memory Requests  Memory Limits  Age
+    ---------                   ----                ------------  ----------  ---------------  -------------  ---
+    kube-system                 kindnet-cnvtd       100m (2%)     100m (2%)   50Mi (0%)        50Mi (0%)      3d23h
+    kube-system                 kube-proxy-6fh4g    0 (0%)        0 (0%)      0 (0%)           0 (0%)         3d23h
+  Allocated resources:
+    (Total limits may be over 100 percent, i.e., overcommitted.)
+    Resource           Requests   Limits
+    --------           --------   ------
+    cpu                100m (2%)  100m (2%)
+    ...
+  Events:              <none>
+```
+
+### Basic Preemption
+
+Gödel Scheduling System provides basic preemption features that are comparable with the offering of the Kubernetes scheduler.
```

Comment:
It would be better if a comparison of preemption strategies between the Gödel Scheduling System and Kubernetes could be provided.

Reply:
@jasonliu747 good suggestion, godel scheduler comprehensive feature list  will be summarized and added to this repo in the near future